quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BootstrapAdminDistTest.java

```diff
@@ -0,0 +1,98 @@
+@RawDistOnly(reason = "Containers are immutable")
+public class BootstrapAdminDistTest {
+
+    @Test
+    @Launch({ "bootstrap-admin", "user", "--no-prompt" })
+    void failNoPassword(LaunchResult result) {
+        assertTrue(result.getErrorOutput().contains("No password provided"),
+                () -> "The Output:\n" + result.getErrorOutput() + "doesn't contains the expected string.");
+    }
+
+    /**
+    @Test
+    @Launch({ "bootstrap-admin", "user", "--expiration=tomorrow" })
+    void failBadExpiration(LaunchResult result) {
+        assertTrue(result.getErrorOutput().contains("Invalid value for option '--expiration': 'tomorrow' is not an int"),
+                () -> "The Output:\n" + result.getErrorOutput() + "doesn't contains the expected string.");
+    }*/
+
+    @Test
+    @Launch({ "bootstrap-admin", "user", "--username=admin", "--password:env=MY_PASSWORD" })
+    void failEnvNotSet(LaunchResult result) {
+        assertTrue(result.getErrorOutput().contains("Environment variable MY_PASSWORD not found"),
+                () -> "The Output:\n" + result.getErrorOutput() + "doesn't contains the expected string.");
+    }
+
+    @Test
+    @WithEnvVars({"MY_PASSWORD", "admin123"})
+    @Launch({ "bootstrap-admin", "user", "--username=admin", "--password:env=MY_PASSWORD" })
+    void createAdmin(LaunchResult result) {
+        assertTrue(result.getErrorOutput().isEmpty(), result.getErrorOutput());
+    }
+
+    @Test
+    @Launch({ "bootstrap-admin", "service", "--no-prompt" })
+    void failServiceAccountNoSecret(LaunchResult result) {
+        assertTrue(result.getErrorOutput().contains("No client secret provided"),
+                () -> "The Output:\n" + result.getErrorOutput() + "doesn't contains the expected string.");
+    }
+
+    @Test
+    @Launch({ "bootstrap-admin", "service", "--client-id=admin", "--client-secret:env=MY_SECRET" })
+    void failServiceAccountEnvNotSet(LaunchResult result) {
+        assertTrue(result.getErrorOutput().contains("Environment variable MY_SECRET not found"),
+                () -> "The Output:\n" + result.getErrorOutput() + "doesn't contains the expected string.");
+    }
+
+    @Test
+    @WithEnvVars({"MY_SECRET", "admin123"})
+    void createAndUseSericeAccountAdmin(KeycloakDistribution dist) throws Exception {
+        RawKeycloakDistribution rawDist = dist.unwrap(RawKeycloakDistribution.class);
+        CLIResult result = rawDist.run("bootstrap-admin", "service", "--client-id=admin", "--client-secret:env=MY_SECRET");
+
+        assertTrue(result.getErrorOutput().isEmpty(), result.getErrorOutput());
+
+        rawDist.setManualStop(true);
+        rawDist.run("start-dev");
+
+        CLIResult adminResult = rawDist.kcadm(List.of("get", "clients", "--server", "http://localhost:8080/auth", "--realm", "master", "--client", "admin", "--secret", "admin123"));
+
+        System.out.println(adminResult.getOutput() + " " + adminResult.getErrorOutput() + " " + adminResult.exitCode());
```

Comment:
Should there be an assert for output/exit code?

Reply:
That was also part of the missing work - however it's not quite working. Trying the same logic with the user succeeds, whereas this fails with an authorization error.